reco_utils/recommender/lightfm/lightfm_utils.py

```diff
@@ -0,0 +1,217 @@
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+import lightfm
+from lightfm.evaluation import precision_at_k, recall_at_k, auc_score
```

Comment:
You are using the `lightfm.evaluation` methods in this file. I think it would be better, for longer term maintenance of the code, to replace them with the ones from this repo. 
Also `auc_score` is not used here and can be removed.

Reply:
Thanks and removed.
the precision_at_k and recall_at_k are for section 4 as explained above, i'll keep them in else section 4 will takes forever to execute.